minimalkv/fsspecstore.py

```diff
@@ -0,0 +1,209 @@
+import io
+from typing import IO, Iterator, Optional
+from urllib.parse import quote, unquote
+
+from fsspec import AbstractFileSystem
+from fsspec.spec import AbstractBufferedFile
+
+from minimalkv import KeyValueStore
+from minimalkv.net._net_common import LAZY_PROPERTY_ATTR_PREFIX, lazy_property
+
+# The complete path of the key is structured as follows:
+# /Users/simon/data/mykvstore/file1
+# <prefix>                    <key>
+# If desired to be a directory, the prefix should end in a slash.
+
+
+class FSSpecStoreEntry(io.BufferedIOBase):
+    """A file-like object for reading from an entry in an FSSpecStore."""
+
+    def __init__(self, file: AbstractBufferedFile):
+        """
+        Initialize an FSSpecStoreEntry.
+
+        Parameters
+        ----------
+        file: AbstractBufferedFile
+            The fsspec file object to wrap.
+        """
+        self._file = file
+
+    def seek(self, loc, whence=0):
+        """
+        Set current file location.
+
+        Parameters
+        ----------
+        loc: int
+            byte location
+        whence: {0, 1, 2}
+            from start of file, current location or end of file, respectively.
+        """
+        if self.closed():
+            raise ValueError("I/O operation on closed file.")
+        try:
+            return self._file.seek(loc, whence)
+        except ValueError:
+            # Map ValueError to IOError
+            raise IOError
+
+    def tell(self):
+        """Return the current offset as int. Always >= 0."""
+        if self.closed():
+            raise ValueError("I/O operation on closed file.")
+        return self._file.tell()
+
+    def read(self, size: Optional[int] = -1) -> bytes:
+        """Return first ``size`` bytes of data.
+
+        If no size is given all data is returned.
+
+        Parameters
+        ----------
+        size : int, optional, default = -1
+            Number of bytes to be returned.
+
+        """
+        return self._file.read(size)
+
+    def seekable(self):  # noqa
+        return self._file.seekable()
+
+    def readable(self):  # noqa
+        return self._file.readable()
```

Comment:
any reason these are not type-hinted?

(the type hinting in this module generally seems to be a bit inconsistent)

Reply:
True yeah. Made it more consistent.